### PR TITLE
Kubeone provider requirements

### PR DESCRIPTION
**Pull request by WeirdMachine**
Fixes https://github.com/kubermatic/kubeone/issues/829

### CODE REVIEW

**Comment by kron4eg** [comment]
I'd very much prefer if we don't change existing URLs.

File: content/kubeone/master/architecture/requirements/provider_support_matrix/azure/azure.en.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+az login
+```
+
+This command will open in your default browser a window where you can authenticate. After you successful logged in get your subscription ID.
+
+```bash
+az account show --query id -o json
+
+********-****-****-****-************
+```
```
Comment (xmudrii) [inline content/kubeone/master/architecture/requirements/provider_support_matrix/azure/azure.en.md:26]: Can we split the command and output into two blocks? That way users can more easily copy the command (e.g. by using the copy button in the right corner of the block).

File: content/kubeone/master/architecture/requirements/provider_support_matrix/kubevirt/kubevirt.en.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
++++
+title = "KubeVirt"
+date = 2021-02-01T14:46:15+02:00
+weight = 7
+
++++
+
+## KubeVirt
```
Comment (xmudrii) [inline content/kubeone/master/architecture/requirements/provider_support_matrix/kubevirt/kubevirt.en.md:8]: We don't officially support KubeVirt. Does it make sense to have this doc?

File: content/kubeone/master/architecture/requirements/provider_support_matrix/AWS/aws.en.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
++++
+title = "AWS"
+date = 2018-09-20T12:12:15+02:00
+weight = 7
+
++++
+
+## AWS
+
+Ensure that the user used by the machine controller has (at least) the following IAM permissions:
```
[the remark this answers is not in the document]
Comment (WeirdMachine) [reply content/kubeone/master/architecture/requirements/provider_support_matrix/AWS/aws.en.md:10]: I think all sections are about the machine controller and not about the required permissions to set up a cluster. 
Therefore this should made clear for all sections here.

File: content/kubeone/master/architecture/requirements/provider_support_matrix/azure/azure.en.md (hunk 1 above)
Comment (WeirdMachine) [reply content/kubeone/master/architecture/requirements/provider_support_matrix/azure/azure.en.md:26]: I think we do not need the output at all

File: content/kubeone/master/architecture/requirements/provider_support_matrix/kubevirt/kubevirt.en.md (hunk 2 above)
Comment (WeirdMachine) [reply content/kubeone/master/architecture/requirements/provider_support_matrix/kubevirt/kubevirt.en.md:8]: Its referenced in the KKP docs though

**Comment by WeirdMachine** [comment]
> I'd very much prefer if we don't change existing URLs.

@xmudrii suggested to use an alias for the infrastructure management section

File: content/kubeone/master/architecture/requirements/infrastructure_management/_index.en.md [hunk 4]
```diff
@@ -19,7 +22,7 @@ example Terraform configs can be found in the [KubeOne's GitHub repository][terr
 {{% notice note %}}
 The example Terraform configurations are optimized for the CI and might not
 be production-ready out of the box. We advise checking the
-[Production Recommendations]({{< ref "../../cheat_sheets/production_recommendations" >}})
+[Production Recommendations][production-recommendations]
```
Comment (xmudrii) [inline content/kubeone/master/architecture/requirements/infrastructure_management/_index.en.md:25]: You can't use this syntax in the notice blocks. For some reason, it's not rendered properly. Instead, please leave it as it was before.

File: content/kubeone/master/architecture/requirements/provider_support_matrix/AWS/aws.en.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
++++
```
Comment (xmudrii) [inline content/kubeone/master/architecture/requirements/provider_support_matrix/AWS/aws.en.md:1]: Nit: please make sure the directory name is lowercase.

File: content/kubeone/master/architecture/requirements/provider_support_matrix/kubevirt/kubevirt.en.md (hunk 2 above)
Comment (xmudrii) [reply content/kubeone/master/architecture/requirements/provider_support_matrix/kubevirt/kubevirt.en.md:8]: KKP does officially support KubeVirt, but that's not the case for KubeOne. I'm worried that if someone sees this doc, they might think we support KubeVirt. I'm not confident in showcasing something that we never tested before.

File: content/kubeone/master/architecture/requirements/provider_support_matrix/AWS/aws.en.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (xmudrii) [reply content/kubeone/master/architecture/requirements/provider_support_matrix/AWS/aws.en.md]: I would rename the "Cloud Provider" section to "Kubermatic machine-controller". That would make it clear that it's about machine-controller and not an overall preview of permissions requirements for a cloud provider.

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**